### PR TITLE
113 vuecli

### LINKED ISSUE #98: [front] add vue linter
efficient and easy one: [https://eslint.vuejs.org/](https://eslint.vuejs.org/)

### LINKED ISSUE #113: [Front] Use vue-cli and .vue files
Depends on #90 

It's difficult to handle all components inside a single HTML file, template are hard to read as they are defined in string literals. That's why we should start using [vue-cli](https://cli.vuejs.org/) and `.vue` files.

### PR DESCRIPTION
**Pull request by viandoxdev**
this should fix #113, this is a kinda big PR but is pretty much a reorganisation and translation to TS, most of the code stays the same.

### CODE REVIEW

**Comment by viandoxdev** [comment]
Also most of the first files changes you'll see are just default vue cli files (linter configs, base index.html) they don't really need to be reviewed closely, what really matters would be App.vue, all the components and the service.

**Review by yannis-mlgrn — CHANGES_REQUESTED** [review]
I thinks it need more commentaries :)

**Comment by viandoxdev** [comment]
> I thinks it need more commentaries :)

be more specific.
Also this is already the subject of another issue so it should be fixed there.

**Comment by yannis-mlgrn** [comment]
I have seen the files but I don't saw have much commentaries. I would add more commentaries for contribute more easily to scratchy

**Comment by viandoxdev** [comment]
> I have seen the files but I don't saw have much commentaries. I would add more commentaries for contribute more easily to scratchy

tel me which files then, what do you not understand, and think that should be explained, ican't just add comments to everything, it'd be useless.

File: front/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            </div>
+            <div class="app_bottom_horizontal_layout">
+                <div class="app_room_editor_start" @click="roomEditorHidden = false">
+                    <div class="app_room_editor_plus"></div>
+                </div>
+                <message-editor v-model:content="currentTypedMessage" v-on:send-message="onMessageSend($event)" v-on:start-writing="isWriting = true" v-on:stop-writing="isWriting = false"></message-editor>
+            </div>
+        </div>
+    </div>
+</template>
```
Comment (yannis-mlgrn) [inline front/src/App.vue:22]: make a commentarie by parts like for this`<!-- app template  -->`

Comment (viandoxdev) [reply front/src/App.vue:22]: I mean its the `<template>` part, of the App.vue file, this doesn't need comment you just need to know vue.

File: front/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# vfront
+
```
Comment (Benvii) [inline front/README.md:4]: This file replaces `doc/frontend/frontend.md`, you should remove it.

You should also add a little description and maybe a screenshot of the frontend.

File: front/src/App.vue (hunk 1 above)
Comment (Benvii) [reply front/src/App.vue:22]: No comment needed here, it's crystal clear.

**Review by Benvii — DISMISSED** [review]
Tested :
- [X] Front start correctly
- [X] Able to create a room
- [X] Able to join a room
- [X] Messages received between 2 users

File: front/src/App.vue [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    Message,
+    User
+} from './service'
+import MessageEditor from './components/MessageEditor.vue';
+import MessageList from './components/MessageList.vue';
+import ActivityBar from './components/ActivityBar.vue';
+import RoomEditor from './components/RoomEditor.vue';
+import UserList from './components/UserList.vue';
+import RoomList from './components/RoomList.vue';
+import Login from './components/Login.vue';
```
[the remark this answers is not in the document]
Comment (viandoxdev) [reply front/src/App.vue:38]: those are just the imports of App, self explanatory.

File: front/src/components/Login.vue [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+        <div class="login_title">Login</div>
+        <label class="login_label" for="pseudo">pseudo</label>
+        <input v-model="pseudo" class="login_input" name="pseudo" type="text" />
+        <div class="login_submit_wrapper">
+            <button class="login_submit" @click="$emit('login', pseudo)">login</button>
+        </div>
+    </div>
+</div>
+</template>
+
```
[the remark this answers is not in the document]
Comment (viandoxdev) [reply front/src/components/Login.vue:13]: not sure what to add here

File: front/src/components/RoomEditor.vue [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                <label class="room_editor_label" for="room_desc">room description</label>
+                <textarea name="room_desc" class="room_editor_input" v-model="createRoomDescription"></textarea>
+                <div class="room_editor_submit_wrapper">
+                    <button class="room_editor_button room_editor_cancel" @click="close()">cancel</button>
+                    <button class="room_editor_button room_editor_submit" @click="$emit('createRoom', {title: createRoomTitle, description: createRoomDescription})">create</button>
+                </div>
+            </div>
+        </div>    
+    </div>
+</template>
```
[the remark this answers is not in the document]
Comment (viandoxdev) [reply front/src/components/RoomEditor.vue:28]: same here

File: front/README.md (hunk 2 above)
Comment (viandoxdev) [reply front/README.md:4]: so should i remove this README or `doc/frontend/frontend.md` ?

Comment (viandoxdev) [reply front/README.md:4]: so i updated `frontend.md` and linked it in `README.md`

**Comment by viandoxdev** [comment]
updated doc and added comments to App.vue, just need a review

**Comment by viandoxdev** [comment]
[new frontend.md](https://github.com/viandoxdev/scratchy/blob/113_vuecli/doc/frontend/frontend.md)

File: front/src/components/RoomEditor.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (viandoxdev) [reply front/src/components/RoomEditor.vue:28]: welp resolving for now

File: front/src/components/Login.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (viandoxdev) [reply front/src/components/Login.vue:13]: resolving for now

**Comment by viandoxdev** [comment]
@Benvii thought on doc ?

**Review by yannis-mlgrn — APPROVED** [review]

**Pull request merged**